fe/fe-core/src/main/java/org/apache/doris/qe/MysqlConnectProcessor.java

```diff
@@ -195,6 +198,24 @@ private void handleExecute() {
         handleExecute(preparedStatementContext.command, stmtId, preparedStatementContext);
     }
 
+    private String parseRealSql(String origStmt, Map<PlaceholderId, Expression> idExpressionMap) {
+        if (idExpressionMap.isEmpty()) {
+            return origStmt;
+        }
+        // To maintain the original order of replace, sort the entries accordingly.
+        List<Map.Entry<PlaceholderId, Expression>> sortedEntries = new ArrayList<>(idExpressionMap.entrySet());
+        sortedEntries.sort(Comparator.comparingInt(entry -> entry.getKey().asInt()));
+        for (Map.Entry<PlaceholderId, Expression> entry : sortedEntries) {
+            Expression expr = entry.getValue();
+            String value = "";
+            if (!expr.isNullLiteral()) {
```

Comment:
why special treat null value？

Reply:
If expr is a NullLiteral, toString() will return "NULL".